3rd-party-licenses.txt

```diff
@@ -1,8 +1,40 @@
 Third Party Licenses
 ====================
 
-This repository contains multiple copies of the project Lab Project FreeRTOS POSIX released under a MIT License: https://github.com/FreeRTOS/Lab-Project-FreeRTOS-POSIX.
-The Lab Project FreeRTOS POSIX MIT license is:
+freertos_apps includes material from the projects listed below (Third Party
+IP). The original copyright notice and the license under which we received
+such Third Party IP, are set forth below.
+
+
+--------------------------------------------------------------------------
+Overview
+--------------------------------------------------------------------------
+
+Multiple folders contain copies of the project Lab Project FreeRTOS POSIX,
+licensed under the MIT license.
+
+Name:      Lab Project FreeRTOS POSIX
+Version:   <<fill version number of commit id>>
+URL:       https://github.com/FreeRTOS/Lab-Project-FreeRTOS-POSIX
+License:   MIT
+Copyright: Copyright (c) 2019 FreeRTOS
+
+
+The folders microros_nucleo_f446ze_extensions and
+microros_olimex_e407_extensions contain code generated by the
+STM32CubeMX software tool of STMicroelectronics. The generated code is
+licensed by ST under the Ultimate Liberty license SLA0044.
+
+Name:      Code generated by STM32CubeMX
+Version:   <<fill version number of the STM32CubeMX tool>>
```

Comment:
@pablogs9, can you please provide the version number?

Reply:
Done